### DISCUSSION TITLE
AsyncCommand or delegateCommand

### DISCUSSION THREAD

**Discussion by developer9969** [PrismLibrary/Prism#2365] [General]
Hi all
Wondering if anyone has any suggestions on whether we should use the AsyncCommand that comes with the Xamarin Community Toolkit instead of the delegateCommand in a prism application. 

any suggestions?

**Comment by brianlagunas** [comment]
Use which ever you need. I personally will never use an "Async" command, but that is my personal choice. You choose the tool that works for you.

**Reply by developer9969** [reply]
Thanks for your reply @brianlagunas , just trying to see if there are benefits in using it. There are times that a developer is not sure on what is "best" and the reason of my question is for find out whether there are some real benefit in using it,

I am intrigued on your personal choice of not using an "Async" command", if you find some time to eloborate on this point I am sure quite a few people including me will learn something.

**Reply by dansiegel** [reply]
@developer9969 as @brianlagunas mentioned use what you need when/where you need it. That said I'd recommend using ReactiveUI's ReactiveCommand before the XCT AsyncCommand. But to be fair here, I generally recommend starting any XF project with Prism + ReactiveUI + ShinyLib.

**Reply by AmrAlSayed0** [reply]
@dansiegel While on the topic, is there any article or a public repo I can look at for more details on how to use Prism + ReactiveUI? I don't understand how 2 different ways of doing something can be used together. There is "SetProperty" in Prism but "RaiseChanged" in ReactiveUI. Which class to inherit from? BindableBase or ReactiveObject. I'm thinking about adopting this but I have yet to see it done.

**Reply by dansiegel** [reply]
@AmrAlSayed0 please see the Prism.Forms + ReactiveUI [sample](https://github.com/PrismLibrary/Prism-Samples-Forms/tree/master/advanced-topics/ReactiveUI).

**Comment by developer9969** [comment]
@dansiegel  hi Dan thank you for your reply, I would love to have a go at your suggestion, is there any "meaty" open source github repo using this combination  where we can learn from and  I can transition my apps quickly? If you have 2 mins would you describe why you prefer Reactive UI  and Shiny , it would help to put a case together if you like and move to it. Many thanks
